src/Symfony/Component/JsonStreamer/DataModel/Write/CollectionNode.php

```diff
@@ -27,6 +27,7 @@ public function __construct(
         private DataAccessorInterface $accessor,
         private CollectionType $type,
         private DataModelNodeInterface $item,
+        private null|DataModelNodeInterface $key = null,
```

Comment:
```suggestion
        private DataModelNodeInterface $key,
```
Can't we make it mandatory?

Reply:
Yes, done.